### PR TITLE
Implement manip gradient in cc.

### PR DESCRIPTION
**Pull request by siyavash**
Create the Roll op gradient function in `manip_grad.cc` corresponding to the `manip_grad.py`.

### CODE REVIEW

**Comment by rthadur** [comment]
@siyavash can you please add some description to the PR

**Comment by siyavash** [comment]
Sure, updated.

File: tensorflow/cc/gradients/manip_grad.cc [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/* Copyright 2016 The TensorFlow Authors. All Rights Reserved.
```
Comment (suharshs) [inline tensorflow/cc/gradients/manip_grad.cc]: 2020

File: tensorflow/cc/gradients/manip_grad_test.cc [hunk 2]
```diff
@@ -0,0 +1,51 @@
+/* Copyright 2016 The TensorFlow Authors. All Rights Reserved.
```
Comment (suharshs) [inline tensorflow/cc/gradients/manip_grad_test.cc]: 2020

File: tensorflow/cc/gradients/manip_grad_test.cc [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+TEST_F(ManipGradTest, RollGrad) {
+  TensorShape shape({5, 4, 3});
+  auto x = Placeholder(scope_, DT_FLOAT, Placeholder::Shape(shape));
+  auto y = Roll(scope_, x, {2, 1}, {0, 1});
+  RunTest(x, shape, y, shape);
+}
+
+}  // anonymous namespace
+}  // tensorflow namespace
```
Comment (suharshs) [inline tensorflow/cc/gradients/manip_grad_test.cc]: nit: // namespace tensorflow instead of // tensorflow namespace

**Review by suharshs — DISMISSED** [review]

**Comment by suharshs** [comment]
Approved, once those  minor comments are addressed, thanks!

File: tensorflow/cc/gradients/manip_grad.cc (hunk 1 above)
Comment (siyavash) [reply tensorflow/cc/gradients/manip_grad.cc]: Done.

File: tensorflow/cc/gradients/manip_grad_test.cc (hunk 2 above)
Comment (siyavash) [reply tensorflow/cc/gradients/manip_grad_test.cc]: Done.

File: tensorflow/cc/gradients/manip_grad_test.cc (hunk 3 above)
Comment (siyavash) [reply tensorflow/cc/gradients/manip_grad_test.cc]: Updated.

**Review by suharshs — DISMISSED** [review]

File: tensorflow/cc/gradients/manip_grad_test.cc [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/cc/framework/gradient_checker.h"
+#include "tensorflow/cc/ops/array_ops.h"
+#include "tensorflow/cc/ops/manip_ops.h"
```
Comment (rthadur) [inline tensorflow/cc/gradients/manip_grad_test.cc]: @siyavash can you please check below error : 
`third_party/tensorflow/cc/gradients/manip_grad_test.cc:18:10: error: module //third_party/tensorflow/cc:gradients_manip_grad_test does not depend on a module exporting 'third_party/tensorflow/cc/ops/manip_ops.h'
see http://go/cpp-features#layering_check; to fix run:
build_cleaner //third_party/tensorflow/cc:gradients_manip_grad_test
#include "third_party/tensorflow/cc/ops/manip_ops.h"`

**Review by rthadur — CHANGES_REQUESTED** [review]

File: tensorflow/cc/gradients/manip_grad_test.cc (hunk 4 above)
Comment (siyavash) [reply tensorflow/cc/gradients/manip_grad_test.cc]: cc_ops (which contains manip_ops) was not included in gradients_manip_grad_test deps, but the Roll op would had been resolved from //tensorflow/core:testlib. I added cc_ops to the deps, but how can I check everything is ok now ?

**Comment by siyavash** [comment]
@suharshs 
How long does it usually take for status to be reported ?!

**Review by suharshs — APPROVED** [review]

**Review by rthadur — APPROVED** [review]

**Pull request merged**